webapp/src/js/services/live-list.js

```diff
@@ -431,55 +434,45 @@ angular.module('inboxServices').factory('LiveList',
       }
       if (removeIndex !== null) {
         idx.list.splice(removeIndex, 1);
-        var removed = idx.dom.splice(removeIndex, 1);
+        const removed = idx.dom[removedItem._id];
+        delete idx.dom[removedItem._id];
 
         $(idx.selector).children().eq(removeIndex).remove();
-        if (removed.length) {
-          return removed[0];
-        }
+        return removed;
       }
     }
 
     function _setSelected(listName, _id) {
-      var i, len, doc,
-          idx = indexes[listName],
-          list = idx.list,
-          previous = idx.selected;
+      const idx = indexes[listName],
+            list = idx.list,
```

Comment:
looks like `list` is only used once, so perhaps clearer to reference it directly as `idx.list` at the time of usage.